### DISCUSSION TITLE
How to avoid "429 Client Error: Too Many Requests for url"

### DISCUSSION THREAD

**Discussion by zsarge** [pndurette/gTTS#325] [Show and tell]
I am currently using gTTS to create an audiobook from hundreds of text files with thousands of characters each.

Having a delay of *5 minutes or more* between calls seems sustainable for hours on end.

I have not done extensive testing so there could be a slightly lower limit.

However, if you are looking for a rough amount of time to delay between large calls, try at least 5 minutes.

**Comment by pazoff** [comment]
Instead of using google text-to-speach you could do your audio books with **Mimic 3**
_A fast, privacy-focused, open-source, neural Text to Speech (TTS) engine running locally on your computer._

It is faster, the voices are good and you can process large volumes of PDF texts with no restrictions.
I have written a small python library to convert my PDF books: [pdf-to-mp3](https://github.com/pazoff/pdf-to-mp3)
